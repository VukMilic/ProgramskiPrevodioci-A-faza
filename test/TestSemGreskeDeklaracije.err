ERROR 17:35:26,944 - Greska: Nije pronadjen tip TypeError u tabeli simbola!  na liniji 3
ERROR 17:35:26,945 - Greska: promenljiva i je vec deklarisana!  na liniji 5
ERROR 17:35:26,945 - Greska: ocekivan tip konstante con1 je Bool!  na liniji 8
ERROR 17:35:26,946 - Greska: konstanta con1 je vec deklarisana!  na liniji 10
ERROR 17:35:26,946 - Greska: Nije pronadjen tip TypeError u tabeli simbola!  na liniji 14
ERROR 17:35:26,947 - Greska: Simbol nepostojeca koji se koristi nije deklarisan!  na liniji 17
ERROR 17:35:26,947 - Greska: tipovi se ne poklapaju prilikom dodele vrednosti!  na liniji 17

ERROR 17:35:26,948 - Parsiranje NIJE uspesno zavrseno!
