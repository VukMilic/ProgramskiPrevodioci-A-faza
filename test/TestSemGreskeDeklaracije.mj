program MJCompiler
	// greske prilikom deklaracija promenljivih ( nepostojeci tip )( ponovna deklaracija )
	TypeError i1;
	int i;
	char i;
	
	// greske prilikom deklaracija konstanti ( dodeljen pogresan tip )( ponovna deklaracija )
	const int con1 = false;
	const Bool con1 = true;
	const char con1 = "H";

{
	// greska prilikom deklaracija lokalnih promenljivih ( nepostojeci tip )( ponovna deklaracija )
	void main() TypeError i1; int i;
	{
	}
}