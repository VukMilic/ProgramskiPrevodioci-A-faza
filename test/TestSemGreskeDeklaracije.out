INFO  17:35:26,881 - Compiling source file: C:\Users\Vuk\Desktop\Finalna Verzija\workspace\MJCompiler\test\program.mj
INFO  17:35:26,921 - #2 program
INFO  17:35:26,921 - #48 MJCompiler
INFO  17:35:26,921 - #48 TypeError
INFO  17:35:26,922 - #48 i1
INFO  17:35:26,923 - #7 ;
INFO  17:35:26,923 - #48 int
INFO  17:35:26,924 - #48 i
INFO  17:35:26,924 - #7 ;
INFO  17:35:26,924 - #48 char
INFO  17:35:26,924 - #48 i
INFO  17:35:26,924 - #7 ;
INFO  17:35:26,924 - #5 const
INFO  17:35:26,924 - #48 int
INFO  17:35:26,924 - #48 con1
INFO  17:35:26,924 - #6 =
INFO  17:35:26,924 - #50 false
INFO  17:35:26,924 - #7 ;
INFO  17:35:26,925 - #5 const
INFO  17:35:26,925 - #48 bool
INFO  17:35:26,925 - #48 con1
INFO  17:35:26,925 - #6 =
INFO  17:35:26,927 - #50 true
INFO  17:35:26,927 - #7 ;
INFO  17:35:26,927 - #5 const
INFO  17:35:26,927 - #48 char
INFO  17:35:26,927 - #48 con1
INFO  17:35:26,927 - #6 =
INFO  17:35:26,927 - #49 'H'
INFO  17:35:26,927 - #7 ;
INFO  17:35:26,927 - #3 {
INFO  17:35:26,927 - #14 void
INFO  17:35:26,928 - #48 main
INFO  17:35:26,928 - #12 (
INFO  17:35:26,928 - #13 )
INFO  17:35:26,929 - #48 TypeError
INFO  17:35:26,930 - #48 i1
INFO  17:35:26,930 - #7 ;
INFO  17:35:26,930 - #48 int
INFO  17:35:26,931 - #48 i
INFO  17:35:26,931 - #7 ;
INFO  17:35:26,931 - #3 {
INFO  17:35:26,931 - #48 i
INFO  17:35:26,931 - #6 =
INFO  17:35:26,931 - #48 nepostojeca
INFO  17:35:26,932 - #7 ;
INFO  17:35:26,933 - #4 }
INFO  17:35:26,934 - #4 }
INFO  17:35:26,940 - Program(
  ProgName(
   MJCompiler
  ) [ProgName]
  DeclarationsList(
    DeclarationsList(
      DeclarationsList(
        DeclarationsList(
          DeclarationsList(
            DeclarationsList(
              NoDeclList(
              ) [NoDeclList]
              DeclGVar(
                GlobalVarDecl(
                  Type(
                   TypeError
                  ) [Type]
                  OneGlobalVarType(
                    GlobalVarIdent(
                     i1
                      NoSquareBrackets(
                      ) [NoSquareBrackets]
                    ) [GlobalVarIdent]
                  ) [OneGlobalVarType]
                ) [GlobalVarDecl]
              ) [DeclGVar]
            ) [DeclarationsList]
            DeclGVar(
              GlobalVarDecl(
                Type(
                 int
                ) [Type]
                OneGlobalVarType(
                  GlobalVarIdent(
                   i
                    NoSquareBrackets(
                    ) [NoSquareBrackets]
                  ) [GlobalVarIdent]
                ) [OneGlobalVarType]
              ) [GlobalVarDecl]
            ) [DeclGVar]
          ) [DeclarationsList]
          DeclGVar(
            GlobalVarDecl(
              Type(
               char
              ) [Type]
              OneGlobalVarType(
                GlobalVarIdent(
                 i
                  NoSquareBrackets(
                  ) [NoSquareBrackets]
                ) [GlobalVarIdent]
              ) [OneGlobalVarType]
            ) [GlobalVarDecl]
          ) [DeclGVar]
        ) [DeclarationsList]
        DeclConst(
          BoolConstDecl(
            Type(
             int
            ) [Type]
            OneConstTypeBool(
             con1
             false
            ) [OneConstTypeBool]
          ) [BoolConstDecl]
        ) [DeclConst]
      ) [DeclarationsList]
      DeclConst(
        BoolConstDecl(
          Type(
           bool
          ) [Type]
          OneConstTypeBool(
           con1
           true
          ) [OneConstTypeBool]
        ) [BoolConstDecl]
      ) [DeclConst]
    ) [DeclarationsList]
    DeclConst(
      CharConstDecl(
        Type(
         char
        ) [Type]
        OneConstTypeChar(
         con1
         'H'
        ) [OneConstTypeChar]
      ) [CharConstDecl]
    ) [DeclConst]
  ) [DeclarationsList]
  MethodDeclarations(
    NoMethodDecl(
    ) [NoMethodDecl]
    MethodDeclaration(
      MethVoidName(
       main
      ) [MethVoidName]
      NoFormParam(
      ) [NoFormParam]
      VarDeclarations(
        VarDeclarations(
          NoVarDecl(
          ) [NoVarDecl]
          VarDecl(
            Type(
             TypeError
            ) [Type]
            OneVarType(
             i1
              NoSquareBrackets(
              ) [NoSquareBrackets]
            ) [OneVarType]
          ) [VarDecl]
        ) [VarDeclarations]
        VarDecl(
          Type(
           int
          ) [Type]
          OneVarType(
           i
            NoSquareBrackets(
            ) [NoSquareBrackets]
          ) [OneVarType]
        ) [VarDecl]
      ) [VarDeclarations]
      Statements(
        NoStmt(
        ) [NoStmt]
        Assignment(
          DesignatorAssignOp(
            DesignatorIdent(
             i
            ) [DesignatorIdent]
            AssignOp(
            ) [AssignOp]
            Expression(
              MinusNot(
              ) [MinusNot]
              Terminal(
                FactorDesignator(
                  DesignatorIdent(
                   nepostojeca
                  ) [DesignatorIdent]
                ) [FactorDesignator]
                NoMulopFactList(
                ) [NoMulopFactList]
              ) [Terminal]
              NoAddopTermList(
              ) [NoAddopTermList]
            ) [Expression]
          ) [DesignatorAssignOp]
        ) [Assignment]
      ) [Statements]
    ) [MethodDeclaration]
  ) [MethodDeclarations]
) [Program]
INFO  17:35:26,940 - ===================================
ERROR 17:35:26,944 - Greska: Nije pronadjen tip TypeError u tabeli simbola!  na liniji 3
INFO  17:35:26,944 - Deklarisana globalna promenljiva i na liniji 4, objektni cvor: Var i: Int, 0, 0
ERROR 17:35:26,945 - Greska: promenljiva i je vec deklarisana!  na liniji 5
ERROR 17:35:26,945 - Greska: ocekivan tip konstante con1 je Bool!  na liniji 8
INFO  17:35:26,946 - Definisana konstanta con1 na liniji 9, objektni cvor: Con con1: Bool, 1, 0
ERROR 17:35:26,946 - Greska: konstanta con1 je vec deklarisana!  na liniji 10
ERROR 17:35:26,946 - Greska: Nije pronadjen tip TypeError u tabeli simbola!  na liniji 14
INFO  17:35:26,947 - Deklarisana lokalna promenljiva i na liniji 14, objektni cvor: Var i: Int, 0, 1
INFO  17:35:26,947 - Simbol i se koristi na liniji 17, objektni cvor: Var i: Int, 0, 1
ERROR 17:35:26,947 - Greska: Simbol nepostojeca koji se koristi nije deklarisan!  na liniji 17
ERROR 17:35:26,947 - Greska: tipovi se ne poklapaju prilikom dodele vrednosti!  na liniji 17
INFO  17:35:26,948 - ===================================
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, -1, -1 
Prog MJCompiler: notype, 0, 1 
   Var i: int, 0, 0 
   Con con1: bool, 1, 0 
   Meth main: notype, 0, 0 
      Var i: int, 0, 1 



ERROR 17:35:26,948 - Parsiranje NIJE uspesno zavrseno!
